javascript/code-challenges/challenges-13.test.js

```diff
@@ -0,0 +1,368 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named longestString that takes in an array of strings and returns the index position of the longest string. 
+------------------------------------------------------------------------------------------------ */
+
+
+const longestString = (arr) => {
+
+  let longest = '';
+  let indexPos = 0;
+  let length = 0;
+
+if (arr.length == 0) {
+  return -1;
+} else {
+  
+for (let i = 0; i < arr.length; i +=1){
+  if (arr[i].length > length) {
+  length = arr[i].length;
+  longest = arr[i];
+  indexPos=i;  
+} 
+}
+return indexPos;}
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named firstLetters that takes in an array of strings and returns an array containing only the first letter of each string.
+
+For example, ['this is great :)', 'wow', 'whyyyyyy :(', ':)))))'] returns ['t', 'w', 'w', ':']
+------------------------------------------------------------------------------------------------ */
+// https://stackoverflow.com/questions/2683466/java-string-indexof-and-empty-strings
+//and 
+//https://stackoverflow.com/questions/10865025/merge-flatten-an-array-of-arrays
+const firstLetters = (arr) => {
+  let pushArray = [];
+  let merged = [];
+  // for (let i = 0; i < arr.length; i += 1){
+  //   pushArray.push(arr[i].[0]);
+  // }
+  // return pushArray;
+
+  for (let i = 0; i < arr.length; i +=1){
+    var firstLetter = arr[i].match(/\b(\w)/g); // get a, then, b, then, c all in different arrays
+    // console.log(firstLetter)
+    pushArray.push(firstLetter);
+    // console.log(pushArray);
+  }
+  for (let i = 0; i < arr.length; i +=1){
+    merged = [].concat.apply([], pushArray);
+  }
+  
+  return merged;
+  
+ // There's probably a better way to do this?
```

Comment:
Good work on the challenges!

We can simplify this:

-  We know our input is an array of strings and our ouput needs to be an array as well.
- What if we use map()? We can map over the input array and grab the first letter of each string. The first letter of each string will live at the first index.

See if you can give this a try! 👍